dm/worker/server.go

```diff
@@ -674,21 +676,50 @@ func (s *Server) startWorker(cfg *config.MysqlConfig) error {
 		return err
 	}
 	for _, kv := range resp.Kvs {
-		infos := common.UpstreamSubTaskKeyAdapter.Decode(string(kv.Key))
-		taskName := infos[1]
 		task := string(kv.Value)
-		cfg := config.NewSubTaskConfig()
-		if err = cfg.Decode(task); err != nil {
+		subTaskcfg := config.NewSubTaskConfig()
+		if err = subTaskcfg.Decode(task); err != nil {
+			return err
+		}
+		subTaskcfg.LogLevel = s.cfg.LogLevel
+		subTaskcfg.LogFile = s.cfg.LogFile
+
+		subTaskCfgs = append(subTaskCfgs, subTaskcfg)
+	}
+
+	if len(cfg.RelayBinLogName) == 0 && len(cfg.RelayBinlogGTID) == 0 {
```

Comment:
If the user specify `relay-binlog-name` explicitly in the config, then run this relay task in DM-worker-A for a long time and scheduled to DM-worker-B, this `relay-binlog-name` may not be exist in the upstream MySQL now. If we use it to start the DM-worker-B, it will fail to fetch binlog, right? Should we need to handle this case?

Reply:
update in 79442de